src/daft-distributed/src/pipeline_node/mod.rs

```diff
@@ -0,0 +1,124 @@
+use std::{
+    collections::HashMap,
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+};
+
+use collect::CollectNode;
+use common_daft_config::DaftExecutionConfig;
+use common_error::DaftResult;
+use common_partitioning::PartitionRef;
+use common_treenode::{Transformed, TreeNode, TreeNodeRewriter};
+use daft_logical_plan::{LogicalPlan, LogicalPlanRef};
+use futures::Stream;
+use limit::LimitNode;
+use translate::translate_pipeline_plan_to_local_physical_plans;
+
+use crate::{channel::Receiver, stage::StageContext};
+
+mod collect;
+mod limit;
+mod translate;
+
+pub(crate) trait DistributedPipelineNode: Send + Sync {
+    #[allow(dead_code)]
+    fn name(&self) -> &'static str;
+    #[allow(dead_code)]
+    fn children(&self) -> Vec<&dyn DistributedPipelineNode>;
+    fn start(&mut self, stage_context: &mut StageContext) -> RunningPipelineNode;
+}
+
+pub(crate) struct RunningPipelineNode {
+    result_receiver: Receiver<PartitionRef>,
+}
+
+impl RunningPipelineNode {
+    fn new(result_receiver: Receiver<PartitionRef>) -> Self {
+        Self { result_receiver }
+    }
+
+    pub fn into_inner(self) -> Receiver<PartitionRef> {
+        self.result_receiver
+    }
+}
+
+impl Stream for RunningPipelineNode {
+    type Item = DaftResult<PartitionRef>;
```

Comment:
instead of `PartitionRef` being passed along, let's pass an 

```
enum PipelineOutput {
  Materialized(PartitionRef),
  Task(Task),
  Running(RunningTask)
}


```